### PR TITLE
feat(toolkit): add prompt

### PR DESCRIPTION
**Pull request by mfadler**

### CODE REVIEW

File: unique_toolkit/unique_toolkit/language_model/prompt.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from string import Template
+from typing import Any
+
+from unique_toolkit.language_model.schemas import (
+    LanguageModelSystemMessage,
+    LanguageModelUserMessage,
+)
+
+
+class Prompt:
```
Comment (klcd) [inline unique_toolkit/unique_toolkit/language_model/prompt.py:10]: Can you add a possibility to see what are the required parameters of the template?

File: unique_toolkit/unique_toolkit/language_model/prompt.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Returns:
+            str: The formatted template string.
+
+        Raises:
+            KeyError: If a required parameter in the template is missing from kwargs.
+        """
+        self._content = self._template.substitute(**kwargs)
+        return self._content
+
+    def to_user_msg(self) -> LanguageModelUserMessage:
```
Comment (klcd) [inline unique_toolkit/unique_toolkit/language_model/prompt.py:41]: Should we raise a warning here if there are unfilled template parameters?

Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/prompt.py:41]: An expception would then be thrown when using .substitute. Only .safe_substitute does not throw an exception when parameters have not been filled.

File: unique_toolkit/unique_toolkit/language_model/prompt.py (hunk 1 above)
Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/prompt.py:10]: Yes, I can add a function for that.

File: unique_toolkit/unique_toolkit/language_model/prompt.py (hunk 2 above)
Comment (klcd) [reply unique_toolkit/unique_toolkit/language_model/prompt.py:41]: Ah nice

Comment (klcd) [reply unique_toolkit/unique_toolkit/language_model/prompt.py:41]: Mind renaming to substitute then?

File: unique_toolkit/unique_toolkit/language_model/prompt.py (hunk 1 above)
Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/prompt.py:10]: as discussed, will be added once we need it

**Review by ThePhilAz — APPROVED** [review]

File: unique_toolkit/unique_toolkit/language_model/prompt.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    def content(self):
+        """
+        Returns the formatted content string.
+
+        Returns:
+            str: The formatted content string.
+        """
+        return self._content
+
+    def format(self, **kwargs: Any) -> str:
```
Comment (klcd) [inline unique_toolkit/unique_toolkit/language_model/prompt.py:72]: I would use substitute here

**Review by klcd — APPROVED** [review]

File: unique_toolkit/unique_toolkit/language_model/prompt.py (hunk 3 above)
Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/prompt.py]: ok

**Pull request merged**